src/librustc_trait_selection/traits/error_reporting/suggestions.rs

```diff
@@ -27,14 +27,14 @@ pub trait InferCtxtExt<'tcx> {
     fn suggest_restricting_param_bound(
         &self,
         err: &mut DiagnosticBuilder<'_>,
-        trait_ref: ty::PolyTraitRef<'_>,
+        trait_ref: ty::PolyTraitRef<'tcx>,
         body_id: hir::HirId,
     );
 
     fn suggest_borrow_on_unsized_slice(
         &self,
         code: &ObligationCauseCode<'tcx>,
-        err: &mut DiagnosticBuilder<'tcx>,
```

Comment:
@nikomatsakis were these bounds changes intentional?

Reply:
They're sort of optional I guess. I did them because:

* I needed to add `'tcx` to some signatures.
* I started by adding `'tcx` *everywhere*, but for `DiagnosticBuilder` there is some code where that doesn't work (some callback or other supplies a `DiagnosticBuilder<'_>`)
* So I rewrote all the diagnostic builders to `DiagnosticBuilder<'_>` so that it would be consistent across all the functions.